tests/unit/test_node.py

```diff
@@ -274,29 +275,31 @@ def test_get_state(self):
         Node class.
 
         """
-        node = Node(None, None)  # Head node
+        node = TransformationProxy(Node(None, None))  # Head node
         n1 = node.Define("a", b="c")  # First child node
 
         # Required dictionaries
-        node_dict = {"children": [n1]}
+        node_dict = {"children": [n1.proxied_node]}
         n1_dict = {
             'operation_name': "Define",
             'operation_args': ("a",),
             'operation_kwargs': {"b": "c"},
             'children': []
         }
-
+        # node is of class Node, its state dictionary can be directly accessed.
```

Comment:
This does not hold true, `node` is of class `Proxy`, so the state has to be accessed through the `proxied_node`

Reply:
Very similar to the situation of the callable generator above. `node` is of class `TransformationProxy`, but the dictionary of the proxied node can actually be directly accesed since the proxy returns all attributes and methods of the node. For the sake of clarity I will put it explicitly